NARFO_BE/NARFO_BE/Controllers/MembersController.cs

```diff
@@ -1,126 +1,183 @@
 using NARFO_BE.Models;
 
 namespace NARFO_BE.Controllers
 {
-    [Route("api/[controller]")]
-    [ApiController]
+    [Route("api/Member")]
+    
     public class MembersController : ControllerBase
     {
         private readonly narfoContext _context;
-        List<Members> members = new List<Members>();
-
-        public MembersController(narfoContext context)
+        List<Member> members = new List<Member>();
+        private IConfiguration Config;
+        public MembersController(narfoContext context, IConfiguration config)
         {
+            Config = config;
             _context = context;
-            if (!_context.Members.Any())
-            {
-                Members newMember = new Members();
-                newMember.LastName = "admin";
-                newMember.Username = "admin";
-                newMember.FirstName = "admin";
-                newMember.Password = "admin";
-                _context.Members.AddAsync(newMember);
-                _context.SaveChanges();
-            }
+
         }
         
-        public void ListofMembers(List<Members> members)
-        {
-            this.members = members;
+      
+        private ActionResult<Member> Json(object p) {
+            throw new NotImplementedException();
         }
-        
 
-        // GET: api/Members
-        [HttpGet]
-        public async Task<ActionResult<IEnumerable<Members>>> GetMembers()
+        private string BuildToken(Member user)
         {
-            return await _context.Members.ToListAsync();
-        }
-        public IEnumerable<Members> GetAllMembers()
-        {
-            return members;
+            var key = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(Config["Jwt:Key"]));
+            var creds = new SigningCredentials(key, SecurityAlgorithms.HmacSha256);
+            var claims = new[] {
+        new Claim(JwtRegisteredClaimNames.Sub, user.Username),
+        new Claim(JwtRegisteredClaimNames.Email, user.Email),
+        new Claim(JwtRegisteredClaimNames.Jti, Guid.NewGuid().ToString())
+    };
+            var token = new JwtSecurityToken(Config["Jwt:Issuer"],
+              Config["Jwt:Issuer"], claims,
+              expires: DateTime.Now.AddDays(360),
+              signingCredentials: creds);
+
+            return new JwtSecurityTokenHandler().WriteToken(token);
         }
 
-        // GET: api/Members/5
-        [HttpGet("{id}")]
-        public async Task<ActionResult<Members>> GetMembers(int id)
-        {
-            var members = await _context.Members.FindAsync(id);
+ 
 
-            if (members == null)
+        // POST: /Account/Login
+        [HttpPost("post/login")]
+        public async Task<ActionResult<Member>> Login([FromBody] Member model)
+        {
+            Member user = null;
+            if(model.Email.Any())
+            {
+                user = await _context.Member.FirstOrDefaultAsync(member => member.Email == model.Email && Encryption.VerifyPassword(model.Password, member.Password));
```

Comment:
pretty sure you can combine these two statements with one linq statement so there is no need for this if..else

Reply:
also try and avoid code duplication as much as possible